flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDeserializationSchema.java

```diff
@@ -351,76 +379,187 @@ private DeserializationRuntimeConverter createFallbackConverter(Class<?> valueTy
-		return Timestamp.valueOf(convertToLocalDateTime(mapper, jsonNode));
+		try {
+			return Timestamp.valueOf(convertToLocalDateTime(mapper, jsonNode));
+		} catch (Throwable t) {
+			throw new ParseErrorException("Unable to deserialize timestamp.", t);
+		}
 	}
 
 	private LocalTime convertToLocalTime(ObjectMapper mapper, JsonNode jsonNode) {
 		// according to RFC 3339 every full-time must have a timezone;
 		// until we have full timezone support, we only support UTC;
 		// users can parse their time as string as a workaround
-		TemporalAccessor parsedTime = RFC3339_TIME_FORMAT.parse(jsonNode.asText());
 
-		ZoneOffset zoneOffset = parsedTime.query(TemporalQueries.offset());
-		LocalTime localTime = parsedTime.query(TemporalQueries.localTime());
+		try {
+			TemporalAccessor parsedTime = RFC3339_TIME_FORMAT.parse(jsonNode.asText());
 
-		if (zoneOffset != null && zoneOffset.getTotalSeconds() != 0 || localTime.getNano() != 0) {
-			throw new IllegalStateException(
-				"Invalid time format. Only a time in UTC timezone without milliseconds is supported yet.");
-		}
+			ZoneOffset zoneOffset = parsedTime.query(TemporalQueries.offset());
+			LocalTime localTime = parsedTime.query(TemporalQueries.localTime());
 
-		return localTime;
+			if (zoneOffset != null && zoneOffset.getTotalSeconds() != 0 || localTime.getNano() != 0) {
+				throw new IllegalStateException(
+					"Invalid time format. Only a time in UTC timezone without milliseconds is supported yet.");
+			}
+
+			return localTime;
+		} catch (Throwable t) {
+			throw new ParseErrorException("Unable to deserialize local time.", t);
+		}
 	}
 
 	private Time convertToTime(ObjectMapper mapper, JsonNode jsonNode) {
-		return Time.valueOf(convertToLocalTime(mapper, jsonNode));
+		try {
+			return Time.valueOf(convertToLocalTime(mapper, jsonNode));
+		} catch (Throwable t) {
+			throw new ParseErrorException("Unable to deserialize time.", t);
+		}
 	}
 
 	private DeserializationRuntimeConverter assembleRowConverter(
 		String[] fieldNames,
 		List<DeserializationRuntimeConverter> fieldConverters) {
-		return (mapper, jsonNode) -> {
-			ObjectNode node = (ObjectNode) jsonNode;
-
-			int arity = fieldNames.length;
-			Row row = new Row(arity);
-			for (int i = 0; i < arity; i++) {
-				String fieldName = fieldNames[i];
-				JsonNode field = node.get(fieldName);
-				Object convertField = convertField(mapper, fieldConverters.get(i), fieldName, field);
-				row.setField(i, convertField);
+		try {
+			return (mapper, jsonNode) -> {
+				ObjectNode node = (ObjectNode) jsonNode;
+				int arity = fieldNames.length;
+				Row row = new Row(arity);
+				for (int i = 0; i < arity; i++) {
+					String fieldName = fieldNames[i];
+					JsonNode field = node.get(fieldName);
+					Object convertField = convertField(mapper, fieldConverters.get(i), fieldName, field);
+					row.setField(i, convertField);
+				}
+
+				return row;
+			};
+		} catch (Throwable t) {
```

Comment:
if the exception is from inner field, the exception will duplicate.  
or we just check jsonNode is a valid ObjectNode, and throw exception if it's not? instead of try all process of the conversion?